### PR TITLE
Editor: Fix adding tags prefixed with spaces

### LINKED ISSUE #1836: Editor: tags get dropped if there are extra spaces in front of them
Steps to reproduce:

1. Start a new post at https://wordpress.com/post
1. Add a title and some content
1. Add several tags, enter them like this: type out a tag, hit enter, press space, repeat
1. Click the "Save" link above the Publish button
1. Watch to see if all of the tags that were entered get saved or not

Result: the tags with spaces in front of them get dropped ([54s](https://cloudup.com/cghINRcC_3U)).

tags-get-dropped-if-there-are-extra-spaces

/hat tip @supernovia for reporting the problem in Slack

Original report at https://en.forums.wordpress.com/topic/tags-disappearing-on-new-posts?replies=14

### PR DESCRIPTION
**Pull request by nylen**
Fixes #1836 (adding tags prefixed with spaces causes the API to discard them when saving the post).

wp-admin trims tag names when adding them, so I've taken the same approach here.

### To test

Follow repro steps at #1836.

Also verify that suggestions are still matched in the presence of leading spaces:

- Add a tag
- Press space
- Begin typing the name of an existing tag and verify that suggestions are still matched correctly despite the leading space

### CODE REVIEW

File: client/components/token-field/README.md [hunk 1]
```diff
@@ -19,10 +19,14 @@ The `value` property is handled in a manner similar to controlled form component
...
 - `value` - An array of strings to display as tokens in the field
-- `valueTransform` - Function to call to transform tokens for display.  This is
+- `displayTransform` - Function to call to transform tokens for display.  This is
   used to decode HTML entities embedded in tags - otherwise entities like `&`
   in tag names are double-encoded like `&amp;` (once by the REST API and once
   by React).
+- `saveTransform` - Function to call to transform tokens for saving.  The
+  default is to trim the token value.  This is used to remove leading and
+  trailing spaces from tag names, like wp-admin does.  (Otherwise the REST API
+  won't save them.)
```
Comment (timmyc) [inline client/components/token-field/README.md]: Might also be worth noting that the `saveTransform` method is called on the search input via `_getMatchingSuggestions`

**Comment by timmyc** [comment]
This is testing out good for me - though I think I'm hitting the problem fixed in #2054 quite a bit while testing... can not click an existing tag and have it added.  Though I verified this issue is happening for me on wpcalypso as well in Chrome.

File: client/components/token-field/README.md (hunk 1 above)
Comment (nylen) [inline client/components/token-field/README.md]: Good idea.  Rebased, did this, and added a test for this new behavior.

**Comment by nylen** [comment]
Rebased after landing #2054 so this is easier to test now.  (Also snuck in an extra test case that should have gone in #2054.)

**Comment by aduth** [comment]
Works and tests well for me.

I wonder if there's ever a case where we'd not want to trim the value? Seems like something we could bake in, not necessarily as a prop.

In any case, this is good to merge :+1:

**Comment by nylen** [comment]
It seemed possible that we'd need another transform there in the future.  If that turns out to not be the case we can go back and simplify a little more.

**Pull request merged**